### ISSUE TITLE
osmapiR: an implementation of OpenStreetMap API v0.6 for R 

### ISSUE THREAD

**Issue by jmaspons** [ropensci/software-review#633]
**Date accepted:** 2024-07-03

Submitting Author Name: Joan Maspons
Submitting Author Github Handle: @jmaspons
Repository:  https://github.com/jmaspons/osmapiR
Version submitted: 0.0.0.22
Submission type: Standard
Editor: @adamhsparks
Reviewers: @jonthegeek, @ccamara
Due date for @jonthegeek: 2024-05-20
Due date for @ccamara: 2024-06-11
Archive: TBD
Version accepted: TBD
Language: en

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Type: Package
Package: osmapiR
Title: OpenStreetMap API
Version: 0.0.0.22
Authors@R: 
    person("Joan", "Maspons", , "joanmaspons@gmail.com", role = c("aut", "cre"),
           comment = c(ORCID = "0000-0003-2286-8727"))
Maintainer: Joan Maspons <joanmaspons@gmail.com>
Description: Implements OpenStreetMap API v0.6
    (<https://wiki.openstreetmap.org/wiki/API_v0.6>).
License: GPL (>= 3)
URL: https://jmaspons.github.io/osmapiR https://github.com/jmaspons/osmapiR
BugReports: https://github.com/jmaspons/osmapiR/issues
Imports: 
    curl,
    httr2,
    xml2
Suggests: 
    httptest2,
    httpuv,
    knitr,
    rmarkdown,
    testthat (>= 3.0.0)
Config/testthat/edition: 3
Encoding: UTF-8
Roxygen: list(markdown = TRUE)
RoxygenNote: 7.3.1
X-schema.org-keywords: open street map, openstreetmap, OSM, openstreetmap-api, osmapi, API
VignetteBuilder: knitr
```

## Scope

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

	- [x] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
        - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [ ] geospatial data
	- [ ] text analysis

- Explain how and why the package falls under these categories (briefly, 1-2 sentences):

`osmapiR` allows retrieving all kind of data from the OpenStreetMap project.

-   Who is the target audience and what are scientific applications of this package?

Useful for people working in OSM and to study the evolution of the map.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

`osmapiR` is the only package to access other OpenStreetMap data than the maps data (map notes, GPS traces, changelogs and users). It can be also useful to get the history of the OSM objects and is the only package that allows editing and upload any kind of data.

To access the OSM map data for purposes other than editing or exploring the history of the objects, perhaps is better to use the other packages that implements the Overpass API ([osmdata](https://github.com/ropensci/osmdata)) or that works with .pbf files ([osmexcract](https://github.com/ropensci/osmextract/)).

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

The package allows accessing all the public data from www.openstreetmap.org. All calls use https and OAuth2 is used for authenticated calls. I redirect the user to osmdata or osmecract in the README and in the documentation of the functions to get map data as recommended by the [OSM policies](https://operations.osmfoundation.org/policies/api/).

-   If you made a pre-submission inquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

-   Explain reasons for any [`pkgcheck` items](https://docs.ropensci.org/pkgcheck/) which your package is unable to pass.

Some functions implement API calls restricted to users with moderator role and are not well documented in the OSM wiki. These functions lack a return value in docs.  Some functions that cause editions in the OSM servers don't have examples to avoid that the user accidentally run it. 

## Technical checks

Confirm each of the following by checking the box.

- [x] I have read the [rOpenSci packaging guide](https://devguide.ropensci.org/building.html).
- [x] I have read the [author guide](https://devdevguide.netlify.app/authors-guide.html) and I expect to maintain this package for at least 2 years or to find a replacement.

This package:

- [x] does not violate the Terms of Service of any service it interacts with.
- [x] has a CRAN and OSI accepted license.
- [x] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme).
- [x] includes [documentation with examples for all functions, created with roxygen2](https://ropensci.github.io/dev_guide/building.html#documentation).
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [x] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage.

## Publication options

- [x] Do you intend for this package to go on CRAN?
- [ ] Do you intend for this package to go on Bioconductor?

- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:

<details>
<summary>MEE Options</summary>

- [ ] The package is novel and will be of interest to the broad readership of the journal.
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [x] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.org/code-of-conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot check package

**Comment by ldecicco-USGS** [comment]
Thanks for the submission @jmaspons! Once you've fixed the above "X", you can trigger the review-bot to re-check the package.

**Comment by jmaspons** [comment]
All returns are now documented. Some requests return nothing, others an xml that is parsed to a data.frame + package classes

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot check package

**Comment by ldecicco-USGS** [comment]
Sorry for the delay. I'll begin searching for an editor, thanks for the submission!

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot assign @adamhsparks as editor

**Comment by adamhsparks** [comment]
### Editor checks:

- [x] **Fit**: The package meets criteria for [fit](policies.md#fit) and [overlap](policies.md#fit)
- [x] **Automated tests:** Package has a testing suite and is tested via GitHub Actions
- [x] **License:** The package has a CRAN accepted license
- [x] **Repository:** The repository link resolves correctly

---

#### Editor comments

Hi there.

I've found a few issues flagged by the initial editor checks:

##### Missing @examples

You have a few empty `#' @examples` tags that need to be provided

```r
✖ osmapi_gps_traces.R:180: @examples requires a value.
✖ osmapi_gps_traces.R:222: @examples requires a value.
✖ osmapi_gps_traces.R:250: @examples requires a value.
```

#### [`goodpractice::gp`](https://github.com/MangoTheCat/goodpractice/) output

```
── GP osmapiR ───────────────────────────────────────────────────────────────────────────────────────────────────────

It is good practice to

  ✖ write unit tests for all functions, and all package code in general. 86% of code lines are
    covered by test cases.

    R/osm_get_changesets.R:97:NA
    R/osm_get_changesets.R:98:NA
    R/osm_get_changesets.R:99:NA
    R/osm_get_objects.R:65:NA
    R/osm_get_objects.R:68:NA
    ... and 251 more lines

  ✖ avoid long code lines, it is bad for readability. Also, many people prefer editor windows
    that are about 80 characters wide. Try make your lines shorter than 80 characters

    inst/httptest2/redact.R:19:81
    inst/httptest2/redact.R:20:81
    inst/httptest2/redact.R:31:81
    R/oauth2.R:28:81
    R/oauth2.R:29:81
    ... and 1330 more lines

  ✖ avoid sapply(), it is not type safe. It might return a vector, or a list, depending on the
    input data. Consider using vapply() instead.

    tests/testthat/test-osmapiR_objects.R:60:21

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

**Reviewers** note highlighted coverage gaps and please inspect what is not covered to see if you think important areas are missed. 

#### `devtools::spell_check()`

You might consider adding the 'Language' field to your DESCRIPTION file.

```
DESCRIPTION does not contain 'Language' field. Defaulting to 'en-US'.
No spelling errors found.
```

#### Comments on Documentation

I noted the use of `''`, I would use `dQuote{}`, myself, and `<code>`, here I would use ` `` ` to denote code blocks, among other things in the ROxygen headers.
I would suggest reviewing the ROxygen documentation <https://roxygen2.r-lib.org/index.html> and possibly enabling Markdown support for formatting the Rd output.

The README has a misspelled R Package name, it should be ({osmextract})[https://github.com/ropensci/osmextract].
I'd also suggest linking to the other OSM packages, _e.g._ {osmdata}, referred to in the README rather than just leaving them as plain text.

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Happy to start looking for reviewers. I recommend using this time to try and address some of the issues raised so they can be checked during review rather that be reflagged.

Let me know if anything is unclear!

**Comment by adamhsparks** [comment]
@ropensci-review-bot assign jonthegeek as reviewer

**Comment by jonthegeek** [comment]
I'm so sorry for the delay! I presented at NYR this past week. I'm finally digging into this today!

**Comment by jonthegeek** [comment]
@ropensci-review-bot check package

**Comment by maurolepore** [comment]
@ropensci-review-bot check package

**Comment by jonthegeek** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- **Briefly describe any working relationship you have (had) with the package authors.**
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [ ] **Vignette(s):** demonstrating major functionality that runs successfully locally
- [ ] **Function Documentation:** for all exported functions
- [ ] **Examples:** (that run successfully locally) for all exported functions
- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [ ] **Functionality:** Any functional claims of the software have been confirmed.
- [ ] **Performance:** Any performance claims of the software have been confirmed.
- [x] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 1 hr (so far)

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

---

### Review Comments

This package looks great!
There's a lot of functionality here to dig into!

A few things to clean up:

These functions are still missing `@examples` in their roxygen blocks: `osm_create_gpx`, `osm_delete_gpx`, `osm_redaction_object`, `osm_update_gpx`.
I know that was discussed previously, but you need to remove the `@examples` roxygen tag if it is empty.
If at all possible, I recommend checking the [Examples section of R Packages (2e)](https://r-pkgs.org/man.html#sec-man-examples) for ideas about how to show *something* for these functions.
Users often scroll down to the examples before reading documentation, so having *something* there to show how the code is intended to work can be helpful, even if it's somewhat contrived (and/or can't actually run).

When I run the tests, there is a lot of "unexpected" output that makes it difficult to see what is happening.
It looks like everything is working in the end, but it would be helpful if you could capture or suppress any expected output, and eliminate anything that shouldn't happen.
For example, `test-user_data.R` is interrupted by `Error in httr2::req_perform(req) : HTTP 404 Not Found.`
You deal with that error in `osm_get_user_details.R` with a `try()` on lines 67-70.
If you use `tryCatch()`, you could replace those lines and suppress the confusing error message, with something like this:

```r
out <- tryCatch(
  osm_details_user(user_id = user_id, format = format),
  error = function(e) empty_user()
)
```

If, on the other hand, you *intend* for the user to see an error in that case, instead I'd wrap the call in `expect_error()`, so it doesn't trigger confusing prints when the test runs, similar to this:

```r
  with_mock_dir("mock_details_user_empty", {
    expect_error({empty_usr <- osm_get_user_details(user_id = 2)})
  })
```

I haven't tracked down the source of the data.frames that print in the test output, but they make it even harder to follow the tests.
I would definitely prefer for those to be suppressed as well!

If you can clean up those pieces, it will make it much easier to see what is supposed to happen as I work through the review!

**Comment by adamhsparks** [comment]
@ropensci-review-bot assign ccamara as reviewer

**Comment by adamhsparks** [comment]
Thank you, @jonthegeek, for your review and thank you @maurolepore for picking up the slack here. I'd missed the conversation on this somehow in my notifications.

@jonthegeek, we'll wait for @ccamara to submit his review so you'll have time to revise yours or add additional thoughts for a while now.

**Comment by jmaspons** [comment]
Thanks for the feedback, @jonthegeek ! I fixed your comments in https://github.com/jmaspons/osmapiR/pull/21 and found a new issue https://github.com/jmaspons/osmapiR/issues/22

@ropensci-review-bot check package

**Comment by ccamara** [comment]
@ropensci-review-bot check package

**Comment by ccamara** [comment]
@adamhsparks There have been some changes since last review, so I wanted to check the package, but I can't do that. I'm reviewing it nevertheless, and may submit review soon, but worth checking anyway?

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot check package

**Comment by ccamara** [comment]
I'm happy to submit my review. Full disclaimer: this is my first package review ever


## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

-   **Briefly describe any working relationship you have (had) with the package authors.**
-   [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor *before* starting your review).

#### Documentation

The package includes all the following forms of documentation:

-   [x] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
-   [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
-   [ ] **Vignette(s):** demonstrating major functionality that runs successfully locally
-   [x] **Function Documentation:** for all exported functions
-   [x] **Examples:** (that run successfully locally) for all exported functions
-   [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

-   [x] **Installation:** Installation succeeds as documented.
-   [ ] **Functionality:** Any functional claims of the software have been confirmed.
-   [ ] **Performance:** Any performance claims of the software have been confirmed.
-   [ ] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
-   [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 3

-   [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

------------------------------------------------------------------------

### Review Comments

I want to thank @jmaspons for their excellent job in doing this useful package which (full disclaimer) I am very likely to be using in the future.

The package implements more than 20 functions and has a function matching every [OSM API call](https://wiki.openstreetmap.org/wiki/API_v0.6#API_calls). All the major functions are well documented, with examples, and have dedicated tests. I couldn't find major issues via automated tools, and I can see that the issues flagged by @jonthegeek's review have been addressed\*.

My main concern is on understanding what this package is (and how it differs from other potentially related packages and OSMs APIs, such as osmdata) and what it can do. While I can significant efforts in documentation functions and readme, I must say that I've been an active OSM contributor since 2014, I'm familiar with all the concepts (changesets, history, gpx traces...) and other APIs, such as overpass, but even with that, I struggle to get a full picture of what this package can do, and more importantly, when should be used. IMHO, this is important given that OSM discourages automated edits. It could be argued that the audience of this package should be familiar with OSM API already, and therefore, they know the answers to those questions, and I'd be OK with that. However, my experience is that more often than not, people are not familiar with most of OSM nuissances and would be looking for a package to "work with OSM" searching on a search engine for "OpenStreetMap R package" or "OSM R package" (or similar queries) and may get here. IMHO, a disclaimer, or note for those people either would be appreciated.

That links to my next item: I find the example on the readme file is useful, well explained, relevant and reproducible, but I wonder how representative it is, given that `osmapiR` can do way more than that. Again, I'd suggest having an indication on how to get started, which can be a link to the aforementioned help page.

So, as a wrap up, I would suggest improving the readme file with the following suggestions:

-   Improve the package description, making it slightly more clearer by extending the succinct "Implements OpenStreetMap API for R." to something like "An R interface to OpenStreetMap API v0.6 for fetching and saving raw geodata from/to the OpenStreetMap database" (copied from <https://wiki.openstreetmap.org/wiki/API>) to clarify the scope of the package and to make it even clearer disctinction with other OSM's APIs (i.e. overpass).

-   Make a note on the scope (when to use or not to use it), and adding any considerations when using it, i.e. mention about automated edits, when putting together disperse information in other parts of the documentation such as API usage (modifying user agent, using a test

-   Outline what the package can do, i.e., by mentioning that can retrieve and update OSM objects, changesets, changeset's discussions, notes, GPS traces and users.

-   Indicate how to get started with the package, beyond the example

That said, I acknowledge that A) most of it has been addressed in the `?osmapiR` , so it could either be adapted from there and indicate prominently that this is the place to get started, and B) I do no think this is a stopper and could be done after the package is reviewed and accepted (I would kindly contribute with the documentation).

\* I didn't notice any warnings when running the tests, and I can see empty examples being deleted, but I can see `osm_create_gpx` . `osm_update_gpx`and `osm_delete_gpx do` not include an empty example, pointing to a dedicated vignette instead, which I believe is missing.

**Comment by adamhsparks** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/633#issuecomment-2121192135 time 1

**Comment by adamhsparks** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/633#issuecomment-2160182185 time 3

**Comment by mpadge** [comment]
Apologies @jmaspons @ccamara and @adamhsparks for checks not appearing previously. The call was issued exactly when our system was in rebuild mode, as mentioned in our [*Dev Guide* (4th point in the list)](https://devguide.ropensci.org/softwarereview_editor.html#upon-submission). I'll manually trigger the check results now.

**Comment by adamhsparks** [comment]
> Apologies @jmaspons @ccamara and @adamhsparks for checks not appearing previously. The call was issued exactly when our system was in rebuild mode, as mentioned in our [_Dev Guide_ (4th point in the list)](https://devguide.ropensci.org/softwarereview_editor.html#upon-submission). I'll manually trigger the check results now.

Thanks, @mpadge. I admit, I didn't read that bullet point before. Now I know!

**Comment by mpadge** [comment]
Another update for @ccamara @adamhsparks. Carlos, you were unable to start the package checks because Adam's' original call to the bot was `assign ccamara as reviewer` when it should have been `assign @ccamara as reviewer`. This logged the reviewer as "ccamara", which of course has no meaning in GitHub terms. I've updated the template above to insert the missing `@` symbols, so now you're officially recognised as a reviewer here. Adam, you can add this to your list of ...
> Now I know!

... even more :rofl:

**Comment by jmaspons** [comment]
Thanks for the review, @ccamara! I will try to improve the README with your feedback and, as you would like to contribute with the docs, I will ask you to review the PR.

> but I can see `osm_create_gpx` . `osm_update_gpx`and `osm_delete_gpx do` not include an empty example, pointing to a dedicated vignette instead, which I believe is missing.

I will check what's missing to include the vignette in the package, but you can check it at https://jmaspons.github.io/osmapiR/articles/how_to_edit_gps_traces.html

**Comment by ccamara** [comment]
@jmaspons a pleasure! Thanks for fixing the vignettes part so quickly (I detected another problem, though: `rmarkdown` is now required to build vignettes, so needs to be specified (see PR https://github.com/jmaspons/osmapiR/pull/24) or added as a required dependency.

Yes, to be honest, when I started reviewing your package yesterday, your name sounded familiar, and I realised that it was a happy coincidence that our paths crossed on OSM a the day before! (For the record: that doesn't mean any conflict of interest in my review).

@mpadge thank you very much for the friendly explanation. You're building such a great community here by treating people so well!

@adamhsparks the experience couldn't have been more positive for me and I'm learning a lot here. I'm not sure if there's anything else I need to do as a reviewer.

**Comment by adamhsparks** [comment]
@ccamara, we wait for @jmaspons to make the requested changes and then the reviews have the chance to comment further or approve

**Comment by jmaspons** [comment]
PR with improvements in the README and docs based on the review of @ccamara at https://github.com/jmaspons/osmapiR/pull/25

**Comment by ccamara** [comment]
@jmaspons , I reviewed the PR and looks great to me. Excellent job.

**Comment by jmaspons** [comment]
> Hey all, I (finally) read an email from @adamhsparks about this. Just to say: would be happy to review. Looks really interesting and can provide feedback if that would be useful.

Should we wait for the review of @Robinlovelace, @adamhsparks  ? He offered to review the package but has not been assigned as a reviewer. I'm not in a hurry, and I'm happy with the feedback. As soon as you give me the OK, I will submit the package to CRAN.

**Comment by adamhsparks** [comment]
@jmaspons, no, Robin was not a reviewer of this package. We have all the reviews in and are waiting on your changes.

**Comment by jmaspons** [comment]
Ok @adamhsparks. From my side, if there is no new feedback from reviewers, I would call it done with https://github.com/jmaspons/osmapiR/pull/21 and https://github.com/jmaspons/osmapiR/pull/25

**Comment by jmaspons** [comment]
Thanks to everybody involved in the review! There is no NEWS file yet as there is no release. I will submit the package after the review as I expect that the planned changes can fit the current API of the package.

### Response to @jonthegeek 

I implemented all the suggestions with https://github.com/jmaspons/osmapiR/pull/21

[x] Reduce output in the tests using expect_snapshot() and the suggested changes to catch expected errors.
[x] Add examples for: `osm_create_gpx`, `osm_delete_gpx` & `osm_update_gpx` in a vignette, and for `osm_redaction_object`.

### Response to @ccamara 

I implemented the suggestions with https://github.com/jmaspons/osmapiR/pull/25

[x] Improve the package description in the README.
[x] Add option to install vignettes when using `remotes::github_install()`.

**Comment by adamhsparks** [comment]
@ropensci-review-bot submit response https://github.com/ropensci/software-review/issues/633#issuecomment-2165983155

**Comment by adamhsparks** [comment]
@jmaspons, can you please submit your response?

**Comment by jmaspons** [comment]
@ropensci-review-bot submit https://github.com/ropensci/software-review/issues/633#issuecomment-2165983155

**Comment by jmaspons** [comment]
@ropensci-review-bot submit response https://github.com/ropensci/software-review/issues/633#issuecomment-2165983155

**Comment by adamhsparks** [comment]
@jmaspons, thank you!

**Comment by adamhsparks** [comment]
@jonthegeek and @ccamara, when you have time, could you approve the changes that have been made or offer further suggestions? 🙏

**Comment by ccamara** [comment]
Hello, I thought I had done it here: https://github.com/ropensci/software-review/issues/633#issuecomment-2163019078 , but I may be missing something. How/Where should I approve any new/same changes? (sorry for the newbie question)

**Comment by jonthegeek** [comment]
## Reviewer Response

#### Final approval (post-review)

- [x] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1.5

**Comment by jonthegeek** [comment]
> Hello, I thought I had done it here: [#633 (comment)](https://github.com/ropensci/software-review/issues/633#issuecomment-2163019078) , but I may be missing something. How/Where should I approve any new/same changes? (sorry for the newbie question).
> 
> Anyway, and just to clarify: I'm happy with the changes being made and the current status of the project.

There's a [template in the guide](https://devguide.ropensci.org/approval2template.html). I had totally missed that, too. Thanks for the poke, @adamhsparks !

**Comment by adamhsparks** [comment]
Thank you, @jonthegeek. @ccamara, John's correct, there's a template and I'm just making sure that I have final approval from the reviewers clearly stated here.

**Comment by adamhsparks** [comment]
Hi folks, I am assuming that this is all good to go. There are just some housekeeping issues left to sort with submitting responses. Can I please get your responses @jmaspons and @ccamara?

**Comment by jmaspons** [comment]
> Logged author response!

Is this enough, @adamhsparks ?

**Comment by adamhsparks** [comment]
Sorry. Yes, that’s on me seeing the bot ask for responses and not realising that @ccamara edited a post

**Comment by adamhsparks** [comment]
@ropensci-review-bot approve osmapiR

**Comment by jmaspons** [comment]
@ropensci-review-bot finalize transfer of jmaspons/osmapiR

**Comment by jmaspons** [comment]
@ropensci-review-bot finalize transfer of osmapiR
